### PR TITLE
Framework: Add RHEL8 GCC OpenMPI configuration

### PR DESCRIPTION
**Pull request by achauphan**
@trilinos/framework 

## Motivation

This build is to be added as a non-blocking PR test to test its stability on RHEL8 machines. 

## Related Issues

- #13070 
- #13067 
- #13074

## Stakeholder Feedback

- [TRILFRAME-647](https://sems-atlassian-son.sandia.gov/jira/browse/TRILFRAME-647)

## Testing

Locally built and tested this build on a RHEL8 machine.

### CODE REVIEW

File: packages/framework/ini-files/config-specs.ini [hunk 1]
```diff
@@ -437,6 +437,8 @@ opt-set-cmake-var SuperLU_INCLUDE_DIRS PATH FORCE : ${SUPERLU_INC|ENV}
 opt-set-cmake-var SuperLU_INCLUDE_DIRS STRING FORCE : ${SUPERLU_INC|ENV}
 opt-set-cmake-var SuperLU_LIBRARY_DIRS PATH FORCE : ${SUPERLU_LIB|ENV}
 opt-set-cmake-var SuperLU_LIBRARY_DIRS STRING FORCE : ${SUPERLU_LIB|ENV}
+opt-set-cmake-var SuperLU_LIBRARY_NAMES STRING FORCE : superlu;m
+opt-set-cmake-var ML_ENABLE_SuperLU BOOL FORCE : OFF
```
Comment (sebrowne) [inline packages/framework/ini-files/config-specs.ini:441]: I just noticed that this breaks the existing GCC 8.3.0 nompi build in my PR.  Probably want to move these two lines into the individual configurations (that's what I did at least).

File: packages/framework/ini-files/config-specs.ini [hunk 2]
```diff
@@ -2949,6 +2951,97 @@ use RHEL7_POST
...
+use NODE-TYPE|SERIAL
+use BUILD-TYPE|DEBUG
+use LIB-TYPE|SHARED
+use KOKKOS-ARCH|NO-KOKKOS-ARCH
+
+use USE-ASAN|NO
+use USE-FPIC|NO
+use USE-MPI|YES
+use USE-PT|NO
+opt-set-cmake-var SEACASAprepro_lib_aprepro_lib_array_test_DISABLE BOOL : ON
```
Comment (sebrowne) [inline packages/framework/ini-files/config-specs.ini:2966]: We previously turned this off for all RHEL7 builds, not sure it needs to be off for RHEL8.  May want to try leaving it on for the new batch and see if that test is still an issue.

**Review by jmlapre — APPROVED** [review]

File: packages/framework/ini-files/config-specs.ini [hunk 3]
```diff
@@ -437,6 +437,8 @@ opt-set-cmake-var SuperLU_INCLUDE_DIRS PATH FORCE : ${SUPERLU_INC|ENV}
 opt-set-cmake-var SuperLU_INCLUDE_DIRS STRING FORCE : ${SUPERLU_INC|ENV}
 opt-set-cmake-var SuperLU_LIBRARY_DIRS PATH FORCE : ${SUPERLU_LIB|ENV}
 opt-set-cmake-var SuperLU_LIBRARY_DIRS STRING FORCE : ${SUPERLU_LIB|ENV}
+opt-set-cmake-var SuperLU_LIBRARY_NAMES STRING FORCE : superlu;m
```
[the remark this answers is not in the document]
Comment (achauphan) [reply packages/framework/ini-files/config-specs.ini]: The `m` is for the `libm` library and is being appended as a list item.

File: packages/framework/ini-files/config-specs.ini [hunk 4]
```diff
@@ -2949,6 +2949,98 @@ use RHEL7_POST
 use rhel7_sems-gnu-8.3.0-openmpi-1.10.7-openmp_release-debug_static_no-kokkos-arch_no-asan_no-complex_no-fpic_mpi_no-pt_no-rdc_no-uvm_deprecated-on_no-package-enables
 use PACKAGE-ENABLES|ALL
 
+[rhel8_sems-gnu-8.5.0-openmpi-4.1.6-serial_debug_shared_no-kokkos-arch_no-asan_no-complex_no-fpic_mpi_no-pt_no-rdc_no-uvm_deprecated-on_no-package-enables]
+#uses sems-archive modules
```
Comment (sebrowne) [inline packages/framework/ini-files/config-specs.ini:2953]: This line is inaccurate for all of the builds added here, please remove it.  Will approve anyhow, so it can be removed as a separate PR if you feel so inclined (we do need to get this in ASAP, we have two weeks until it has to go live).

**Review by sebrowne — APPROVED** [review]
